lib/galaxy/util/__init__.py

```diff
@@ -1696,26 +1699,28 @@ def stream_to_open_named_file(stream, fd, filename, source_encoding=None, source
     except Exception:
         target_encoding = DEFAULT_ENCODING  # utf-8
     use_source_encoding = source_encoding is not None
-    while True:
-        chunk = stream.read(CHUNK_SIZE)
-        if not chunk:
-            break
-        if use_source_encoding:
-            # If a source encoding is given we use it to convert to the target encoding
-            try:
-                if not isinstance(chunk, str):
-                    chunk = chunk.decode(source_encoding, source_error)
-                os.write(fd, chunk.encode(target_encoding, target_error))
-            except UnicodeDecodeError:
-                use_source_encoding = False
+    try:
```

Comment:
@mvdbeek .. should there be an `except` that closes the file pointer and reraises the exception? .. Searching for the cause of https://github.com/galaxyproject/galaxy/issues/13136

Reply:
You can do try: finally without an except clause:
```
In [3]: try:
   ...:     x = 1
   ...:     raise Exception('boo')
   ...: finally:
   ...:     x = 2
   ...:
---------------------------------------------------------------------------
Exception                                 Traceback (most recent call last)
<ipython-input-3-778dedc359d2> in <module>
      1 try:
      2     x = 1
----> 3     raise Exception('boo')
      4 finally:
      5     x = 2

Exception: boo

In [4]: x
Out[4]: 2
```
I'd suggest checking the stack with `py-spy dump --pid <pid_of_gx_process>`, that should give you an indication on what Galaxy is doing while it is stuck

Reply:
Seems that `Tabular.set_meta` is the culprit here. Also https://github.com/galaxyproject/galaxy/pull/13137 solves the problem. 

But I would like to understand what is going on.

Reply:
You have an input with failed metadata, so that job is just sitting in the job handlers and is never getting dispatched, I'd assume.